server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -149,7 +152,27 @@ object InteractiveSession extends Logging {
     builderProperties ++= conf
 
     def livyJars(livyConf: LivyConf, scalaVersion: String): List[String] = {
-      Option(livyConf.get(LIVY_REPL_JARS)).map(_.split(",").toList).getOrElse {
+      Option(livyConf.get(LIVY_REPL_JARS)).map { jars =>
+        val regex = """[\w-]+_(\d\.\d\d).*\.jar""".r
+        jars.split(",").flatMap { n =>
+          val trimmed = n.trim
+          if (!trimmed.startsWith("local")) {
+            // expand the glob path for file and hdfs scheme
+            val path = new Path(toURI(trimmed))
+            val pathFs = FileSystem.get(path.toUri, livyConf.hadoopConf)
```

Comment:
I don't think the latency will be huge by adding this API. 

1. `FileSystem` is locally cached, each time when you call `FileSystem.get` it will not create a new instance, just fetch a cached one.
2. For the latency of `Path.globStatus`, local hdfs client will cache the metadata of NN, so it may not trigger a RPC for each call. And to the worse, only one RPC call is enough. The latency is not so huge.
3. Compare to the time of allocating containers and starting Spark applications on YARN, the time spent on this part is very very small. Usually it will take about 30+ seconds to fully start a Spark application on YARN (in our test), here this RPC call may only take less than 1 second.